synapse/appservice/scheduler.py

```diff
@@ -148,6 +149,16 @@ def enqueue_for_appservice(
         ):
             return
 
+        if events:
+            # XXX: Special patch just for Gitter which we should remove after the import,
+            # https://github.com/matrix-org/synapse/pull/14729
+            #
+            # Ignore events that come from our own users. We probably already know about
+            # them and sent them ourself.
+            events = [
+                event for event in events if not self._hs.is_mine_id(event.sender)
+            ]
```

Comment:
Update: @turt2live shared a similar t2bot.io patch that they've been running for a long-time: https://github.com/t2bot/synapse/commit/3c1cd9ace85262596523e4316b10b1ead2b5dec5